components/scream/scripts/utils.py

```diff
@@ -366,7 +366,8 @@ def merge_git_ref(git_ref, repo=None):
     """
     Merge given git ref into the current branch, and updates submodules
     """
-    run_cmd_no_fail("git merge {} -m 'Automatic merge of {}'".format(git_ref,git_ref),from_dir=repo)
+    expect(is_repo_clean(), "Cannot merge ref '{}'. The repo is not clean.".format(git_ref))
+    stat = run_cmd("git merge {} -m 'Automatic merge of {}'".format(git_ref,git_ref),from_dir=repo)
```

Comment:
This should be run_cmd_no_fail I think. That, or the contents of `stat` should be checked.

Reply:
True. Thanks.